newrelic-agent/src/test/java/com/newrelic/agent/util/VoidBlockingQueueTest.java

```diff
@@ -0,0 +1,299 @@
+/*
+ *
+ *  * Copyright 2024 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package com.newrelic.agent.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * VoidBlockingQueue is not a very interesting class to test.
+ * These tests are meant to fail if someone changes the functionality of the class.
+ */
+public class VoidBlockingQueueTest {
+    @Test
+    public void add() {
+        BlockingQueue<Object> blockingQueue = VoidBlockingQueue.getInstance();
+        assertTrue(blockingQueue.add(new Object()));
+        assertTrue(blockingQueue.isEmpty());
+    }
+
+    @Test
+    public void offer() {
+        BlockingQueue<Object> blockingQueue = VoidBlockingQueue.getInstance();
+        assertTrue(blockingQueue.offer(new Object()));
+        assertTrue(blockingQueue.isEmpty());
+    }
+
+    @Test
+    public void remove() {
```

Comment:
Simplify the tests a tiny bit if you use `@Test(expected = NoSuchElementException.class)`

```
public  void remove() {
    BlockingQueue<Object> blockingQueue = VoidBlockingQueue.getInstance();
    blockingQueue.remove();
}
```